homeassistant/components/sanix/sensor.py

```diff
@@ -0,0 +1,148 @@
+        key=ATTR_API_TIME,
+        device_class=SensorDeviceClass.TIMESTAMP,
+        translation_key="time",
+    ),
+    SanixSensorEntityDescription(
+        key=ATTR_API_FILL_PERCENTAGE,
+        native_unit_of_measurement=PERCENTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+        translation_key="fill_percentage",
+        icon="mdi:water-percent",
+    ),
+    SanixSensorEntityDescription(key=ATTR_API_SSID, translation_key="ssid"),
+    SanixSensorEntityDescription(key=ATTR_API_DEVICE_NO, translation_key="device_no"),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Sanix Sensor entities based on a config entry."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+
+    sensors = []
+    for description in SENSOR_TYPES:
+        if coordinator.data.get(description.key):
+            sensors.append(
+                SanixSensor(coordinator, entry.title, str(entry.unique_id), description)
+            )
+
+    async_add_entities(sensors)
+
+
+class SanixSensor(CoordinatorEntity[SanixCoordinator], SensorEntity):
+    """Sanix Sensor entity."""
+
+    _attr_has_entity_name = True
+    entity_description: SanixSensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: SanixCoordinator,
+        name: str,
+        serial_no: str,
+        description: SanixSensorEntityDescription,
+    ) -> None:
+        """Pass coordinator to CoordinatorEntity."""
+        super().__init__(coordinator)
+        self._attr_unique_id = f"{name}-{description.key}".lower()
+        self._attr_native_value = self.format_value(
+            coordinator.data[description.key], description.key
+        )
+        self.entity_description = description
+
+        self._attr_device_info = DeviceInfo(
+            entry_type=DeviceEntryType.SERVICE,
+            identifiers={(DOMAIN, serial_no)},
+            manufacturer=MANUFACTURER,
+            name=name,
+            serial_number=serial_no,
+        )
+
+    def format_value(self, value, key):
+        """Format the value to the native format."""
+        try:
+            if key == ATTR_API_SERVICE_DATE:
+                value = datetime.datetime.strptime(value, "%d.%m.%Y").date()
+            elif key == ATTR_API_TIME:
+                value = datetime.datetime.strptime(value, "%d.%m.%Y %H:%M:%S").replace(
+                    tzinfo=ZoneInfo("Europe/Warsaw")
+                )
+        except ValueError:
+            _LOGGER.warning(
+                "Could not format the '%s' sensor. Retrieved value: %s", key, value
+            )
+            return None
+        return value
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        self._attr_native_value = self.format_value(
+            self.coordinator.data[self.entity_description.key],
+            self.entity_description.key,
+        )
```

Comment:
Please put this in a separate function and you can call this from the constructor

Reply:
Do you mean something like it is now?

Reply:
create a function _update_attr() in which you only set the `self._attr_*`. This will be called from the _handle_coordinator_update and from the constructor to dont repeat the setting logic twice

Reply:
got it, thanks :)